from PyQt5 import QtCore, QtGui, QtWidgets
import numpy as np
from scipy import interpolate
from mplwidget import MplWidget

#GUI for Input Menu
class Ui_menuInput(object):
    #Process for calculating somatotype
    def calculateSomatotype(self):
        #Taking Antrophometry value from GUI
        height= float(self.lineEditHeight.text())
        weight= float(self.lineEditWeight.text())
        tricepSF= float(self.lineEditTricepSF.text())
        subscapularSF= float(self.lineSubscapularSF.text())
        medialCalfSF= float(self.lineEditMedialCalfSF.text())
        supraspinaleSF= float(self.lineEditSupraspinaleSF.text())
        boFemur= float(self.lineEditBoFemur.text())
        boHumerus= float(self.lineEditBoFemur.text())
        tricepG= float(self.lineEditTricepG.text())
        calfG= float(self.lineEditCalfG.text())
        #Calculating with Heath-Carter formula
        global endomorph, mesomorph, ectomorph, dispJenisSoma
        tempEndo=(tricepSF+subscapularSF+supraspinaleSF)*170.18/height
        endomorph=-0.7182+(0.145*tempEndo)-(0.00068*tempEndo*tempEndo)+(0.0000014*tempEndo*tempEndo*tempEndo)
        mesomorph=(0.858*boHumerus)+(0.601*boFemur)+(0.188*tricepG)+(0.161*calfG)-(0.131*height)+4.5
        tempEcto=height/(pow(weight,1/3))
        if tempEcto>40.74:
            ectomorph=(0.732*tempEcto)-28.58
        elif tempEcto<40.74 and tempEcto>39.65:
            ectomorph=(0.463*tempEcto)-17.615
        else:
            ectomorph=0.5
        dispJSoma={endomorph:"Endomorph",mesomorph:"Mesomorph",ectomorph:"Ectomorph"}
        dispJenisSoma=dispJSoma.get(max(dispJSoma))

    #Process for opening Output Menu Window
    def openMenuOutput(self):
        self.calculateSomatotype()
        self.window=QtWidgets.QMainWindow()
        self.ui =Ui_MenuOutput()
        self.ui.setupUi(self.window)
        menuInput.hide()
        self.window.show()
        global closeMenuOutput
        def closeMenuOutput():
            self.window.hide()

    def setupUi(self, menuInput):
        menuInput.setObjectName("menuInput")
        menuInput.setFixedSize(376, 546)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Preferred, QtWidgets.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(menuInput.sizePolicy().hasHeightForWidth())
        menuInput.setSizePolicy(sizePolicy)
        menuInput.setWindowIcon(QtGui.QIcon('icon.png'))
        self.centralwidget = QtWidgets.QWidget(menuInput)
        self.centralwidget.setObjectName("centralwidget")
        self.gridLayout = QtWidgets.QGridLayout(self.centralwidget)
        self.gridLayout.setObjectName("gridLayout")
        self.labelTitle = QtWidgets.QLabel(self.centralwidget)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(21)
        font.setBold(True)
        font.setWeight(75)
        font.setStrikeOut(False)
        font.setKerning(True)
        self.labelTitle.setFont(font)
        self.labelTitle.setAlignment(QtCore.Qt.AlignCenter)
        self.labelTitle.setObjectName("labelTitle")
        self.gridLayout.addWidget(self.labelTitle, 0, 0, 1, 2)
        self.scrollArea = QtWidgets.QScrollArea(self.centralwidget)
        self.scrollArea.setWidgetResizable(True)
        self.scrollArea.setObjectName("scrollArea")
        self.scrollAreaWidgetContents = QtWidgets.QWidget()
        self.scrollAreaWidgetContents.setGeometry(QtCore.QRect(0, 0, 356, 395))
        self.scrollAreaWidgetContents.setObjectName("scrollAreaWidgetContents")
        self.gridLayout_2 = QtWidgets.QGridLayout(self.scrollAreaWidgetContents)
        self.gridLayout_2.setObjectName("gridLayout_2")
        self.labelSubscapularSF = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelSubscapularSF.setFont(font)
        self.labelSubscapularSF.setObjectName("labelSubscapularSF")
        self.gridLayout_2.addWidget(self.labelSubscapularSF, 6, 0, 1, 1)
        self.labelMedialCalfSF = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelMedialCalfSF.setFont(font)
        self.labelMedialCalfSF.setObjectName("labelMedialCalfSF")
        self.gridLayout_2.addWidget(self.labelMedialCalfSF, 7, 0, 1, 1)
        self.lineEditHeight = QtWidgets.QLineEdit(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.lineEditHeight.setFont(font)
        self.lineEditHeight.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.lineEditHeight.setObjectName("lineEditHeight")
        self.gridLayout_2.addWidget(self.lineEditHeight, 0, 1, 1, 1)
        self.labelCalfG = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelCalfG.setFont(font)
        self.labelCalfG.setObjectName("labelCalfG")
        self.gridLayout_2.addWidget(self.labelCalfG, 14, 0, 1, 1)
        self.labelcm5 = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelcm5.setFont(font)
        self.labelcm5.setObjectName("labelcm5")
        self.gridLayout_2.addWidget(self.labelcm5, 14, 5, 1, 1)
        self.labelHeight = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelHeight.setFont(font)
        self.labelHeight.setObjectName("labelHeight")
        self.gridLayout_2.addWidget(self.labelHeight, 0, 0, 1, 1)
        self.labelmm3 = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelmm3.setFont(font)
        self.labelmm3.setObjectName("labelmm3")
        self.gridLayout_2.addWidget(self.labelmm3, 7, 5, 1, 1)
        self.lineEditTHumerus = QtWidgets.QLineEdit(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.lineEditTHumerus.setFont(font)
        self.lineEditTHumerus.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.lineEditTHumerus.setObjectName("lineEditTHumerus")
        self.gridLayout_2.addWidget(self.lineEditTHumerus, 12, 1, 1, 1)
        self.labelcm3 = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelcm3.setFont(font)
        self.labelcm3.setObjectName("labelcm3")
        self.gridLayout_2.addWidget(self.labelcm3, 12, 5, 1, 1)
        self.lineEditSupraspinaleSF = QtWidgets.QLineEdit(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.lineEditSupraspinaleSF.setFont(font)
        self.lineEditSupraspinaleSF.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.lineEditSupraspinaleSF.setObjectName("lineEditSupraspinaleSF")
        self.gridLayout_2.addWidget(self.lineEditSupraspinaleSF, 10, 1, 1, 1)
        self.lineEditBoFemur = QtWidgets.QLineEdit(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.lineEditBoFemur.setFont(font)
        self.lineEditBoFemur.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.lineEditBoFemur.setObjectName("lineEditBoFemur")
        self.gridLayout_2.addWidget(self.lineEditBoFemur, 11, 1, 1, 1)
        self.labelmm4 = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelmm4.setFont(font)
        self.labelmm4.setObjectName("labelmm4")
        self.gridLayout_2.addWidget(self.labelmm4, 10, 5, 1, 1)
        self.lineEditCalfG = QtWidgets.QLineEdit(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.lineEditCalfG.setFont(font)
        self.lineEditCalfG.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.lineEditCalfG.setObjectName("lineEditCalfG")
        self.gridLayout_2.addWidget(self.lineEditCalfG, 14, 1, 1, 1)
        self.lineEditTricepSF = QtWidgets.QLineEdit(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.lineEditTricepSF.setFont(font)
        self.lineEditTricepSF.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.lineEditTricepSF.setObjectName("lineEditTricepSF")
        self.gridLayout_2.addWidget(self.lineEditTricepSF, 4, 1, 1, 1)
        self.labelmm = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelmm.setFont(font)
        self.labelmm.setObjectName("labelmm")
        self.gridLayout_2.addWidget(self.labelmm, 4, 5, 1, 1)
        self.labelLTricepG = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelLTricepG.setFont(font)
        self.labelLTricepG.setObjectName("labelLTricepG")
        self.gridLayout_2.addWidget(self.labelLTricepG, 13, 0, 1, 1)
        self.labelcm = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelcm.setFont(font)
        self.labelcm.setObjectName("labelcm")
        self.gridLayout_2.addWidget(self.labelcm, 0, 5, 1, 1)
        self.labelcm4 = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelcm4.setFont(font)
        self.labelcm4.setObjectName("labelcm4")
        self.gridLayout_2.addWidget(self.labelcm4, 13, 5, 1, 1)
        self.labelSupraspinaleSF = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelSupraspinaleSF.setFont(font)
        self.labelSupraspinaleSF.setObjectName("labelSupraspinaleSF")
        self.gridLayout_2.addWidget(self.labelSupraspinaleSF, 10, 0, 1, 1)
        self.labelTricepSF = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelTricepSF.setFont(font)
        self.labelTricepSF.setObjectName("labelTricepSF")
        self.gridLayout_2.addWidget(self.labelTricepSF, 4, 0, 1, 1)
        self.lineSubscapularSF = QtWidgets.QLineEdit(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.lineSubscapularSF.setFont(font)
        self.lineSubscapularSF.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.lineSubscapularSF.setObjectName("lineSubscapularSF")
        self.gridLayout_2.addWidget(self.lineSubscapularSF, 6, 1, 1, 1)
        self.lineEditTricepG = QtWidgets.QLineEdit(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.lineEditTricepG.setFont(font)
        self.lineEditTricepG.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.lineEditTricepG.setObjectName("lineEditTricepG")
        self.gridLayout_2.addWidget(self.lineEditTricepG, 13, 1, 1, 1)
        self.lineEditMedialCalfSF = QtWidgets.QLineEdit(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.lineEditMedialCalfSF.setFont(font)
        self.lineEditMedialCalfSF.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.lineEditMedialCalfSF.setObjectName("lineEditMedialCalfSF")
        self.gridLayout_2.addWidget(self.lineEditMedialCalfSF, 7, 1, 1, 1)
        self.labelBoFemur = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelBoFemur.setFont(font)
        self.labelBoFemur.setObjectName("labelBoFemur")
        self.gridLayout_2.addWidget(self.labelBoFemur, 11, 0, 1, 1)
        self.labelBoHumerus = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelBoHumerus.setFont(font)
        self.labelBoHumerus.setObjectName("labelBoHumerus")
        self.gridLayout_2.addWidget(self.labelBoHumerus, 12, 0, 1, 1)
        self.labelmm2 = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelmm2.setFont(font)
        self.labelmm2.setObjectName("labelmm2")
        self.gridLayout_2.addWidget(self.labelmm2, 6, 5, 1, 1)
        self.labelcm2 = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelcm2.setFont(font)
        self.labelcm2.setObjectName("labelcm2")
        self.gridLayout_2.addWidget(self.labelcm2, 11, 5, 1, 1)
        self.labelWeight = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelWeight.setFont(font)
        self.labelWeight.setObjectName("labelWeight")
        self.gridLayout_2.addWidget(self.labelWeight, 3, 0, 1, 1)
        self.lineEditWeight = QtWidgets.QLineEdit(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.lineEditWeight.setFont(font)
        self.lineEditWeight.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.lineEditWeight.setObjectName("lineEditWeight")
        self.gridLayout_2.addWidget(self.lineEditWeight, 3, 1, 1, 1)
        self.labelkg = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelkg.setFont(font)
        self.labelkg.setObjectName("labelkg")
        self.gridLayout_2.addWidget(self.labelkg, 3, 5, 1, 1)
        self.scrollArea.setWidget(self.scrollAreaWidgetContents)
        self.gridLayout.addWidget(self.scrollArea, 1, 0, 1, 2)
        self.buttonProcess = QtWidgets.QPushButton(self.centralwidget)
        self.buttonProcess.setMinimumSize(QtCore.QSize(0, 40))
        self.buttonProcess.clicked.connect(self.openMenuOutput)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(10)
        font.setBold(False)
        font.setWeight(50)
        self.buttonProcess.setFont(font)
        self.buttonProcess.setAutoDefault(False)
        self.buttonProcess.setObjectName("buttonProcess")
        self.gridLayout.addWidget(self.buttonProcess, 2, 0, 1, 2)
        menuInput.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(menuInput)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 376, 21))
        self.menubar.setObjectName("menubar")
        menuInput.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(menuInput)
        self.statusbar.setObjectName("statusbar")
        menuInput.setStatusBar(self.statusbar)

        self.retranslateUi(menuInput)
        QtCore.QMetaObject.connectSlotsByName(menuInput)

    def retranslateUi(self, menuInput):
        _translate = QtCore.QCoreApplication.translate
        menuInput.setWindowTitle(_translate("menuInput", "Input Menu"))
        self.labelTitle.setText(_translate("menuInput", "Antrophometry Data"))
        self.labelSubscapularSF.setText(_translate("menuInput", "Subscapular Skinfold"))
        self.labelMedialCalfSF.setText(_translate("menuInput", "Medial Calf Skinfold"))
        self.labelCalfG.setText(_translate("menuInput", "Calf Girth"))
        self.labelcm5.setText(_translate("menuInput", "cm"))
        self.labelHeight.setText(_translate("menuInput", "Height"))
        self.labelmm3.setText(_translate("menuInput", "mm"))
        self.labelcm3.setText(_translate("menuInput", "cm"))
        self.labelmm4.setText(_translate("menuInput", "mm"))
        self.labelmm.setText(_translate("menuInput", "mm"))
        self.labelLTricepG.setText(_translate("menuInput", "Tricep Girth"))
        self.labelcm.setText(_translate("menuInput", "cm"))
        self.labelcm4.setText(_translate("menuInput", "cm"))
        self.labelSupraspinaleSF.setText(_translate("menuInput", "Supraspinale Skinfold"))
        self.labelTricepSF.setText(_translate("menuInput", "Tricep Skinfold"))
        self.labelBoFemur.setText(_translate("menuInput", "Breadth of Femur"))
        self.labelBoHumerus.setText(_translate("menuInput", "Breadth of Humerus"))
        self.labelmm2.setText(_translate("menuInput", "mm"))
        self.labelcm2.setText(_translate("menuInput", "cm"))
        self.labelWeight.setText(_translate("menuInput", "Weight"))
        self.labelkg.setText(_translate("menuInput", "kg"))
        self.buttonProcess.setText(_translate("menuInput", "Process"))


#GUI for Output Menu
class Ui_MenuOutput(object):
    #Process for back to input menu
    def openMenuInput(self):
        closeMenuOutput()
        menuInput.show()

    #Process for drawing somatochart
    def drawSomatochart(self):
        xCoordinate= ectomorph-endomorph
        xCoordinate=round(xCoordinate, 1)
        yCoordinate= 2*mesomorph-(ectomorph+endomorph)
        yCoordinate=round(yCoordinate, 1)
        x=[0,-6,6,0,-4.5,4.5,0]
        y=[12,-6,-6,0,4.5,4.5,-7.5]
        self.MplWidget.canvas.axes.clear()
        self.MplWidget.canvas.axes.scatter(x,y)
        self.MplWidget.canvas.axes.plot([0,0],[12,-7.5], 'bo', linestyle="--")
        self.MplWidget.canvas.axes.plot([-6,4.5],[-6,4.5], 'bo', linestyle="--")
        self.MplWidget.canvas.axes.plot([6,-4.5],[-6,4.5], 'bo', linestyle="--")
        self.MplWidget.canvas.axes.plot([6,-4.5],[-6,4.5], 'bo', linestyle="--")
        nodes = np.array( [ [0, 12], [4.5, 4.5], [6,-6]] )
        xer = nodes[:,0]
        yer = nodes[:,1]
        tck,u     = interpolate.splprep( [xer,yer] ,k =2 )
        xnew,ynew = interpolate.splev( np.linspace( 0, 1, 100 ), tck,der = 0)
        self.MplWidget.canvas.axes.plot(xnew ,ynew,'b',linestyle="--" )
        nodes1 = np.array( [ [6, -6], [0, -7.5], [-6,-6]] )
        xer1 = nodes1[:,0]
        yer1 = nodes1[:,1]
        tck,u     = interpolate.splprep( [xer1,yer1] ,k =2 )
        xnew1,ynew1 = interpolate.splev( np.linspace( 0, 1, 100 ), tck,der = 0)
        self.MplWidget.canvas.axes.plot(xnew1 ,ynew1,'b',linestyle="--" )
        nodes1 = np.array( [ [-6, -6], [-4.5, 4.5], [0,12]] )
        xer1 = nodes1[:,0]
        yer1 = nodes1[:,1]
        tck,u     = interpolate.splprep( [xer1,yer1] ,k =2 )
        xnew1,ynew1 = interpolate.splev( np.linspace( 0, 1, 100 ), tck,der = 0)
        self.MplWidget.canvas.axes.plot(xnew1 ,ynew1,'b',linestyle="--" )
        self.MplWidget.canvas.axes.text(0, 13, 'MESOMORPH', fontsize = 10,horizontalalignment='center')
        self.MplWidget.canvas.axes.text(-6, -8, 'ENDOMORPH', fontsize = 10,horizontalalignment='center')
        self.MplWidget.canvas.axes.text(6, -8, 'ECTOMORPH', fontsize = 10,horizontalalignment='center')
        self.MplWidget.canvas.axes.axis([-8,8,-8,13])
        self.MplWidget.canvas.axes.text(xCoordinate,yCoordinate+0.5, '(%s,%s)'%(xCoordinate,yCoordinate), fontsize = 11,horizontalalignment='center',color='red',weight="bold")
        self.MplWidget.canvas.axes.scatter(xCoordinate,yCoordinate)
        self.MplWidget.canvas.axes.set_facecolor("red")
        self.MplWidget.canvas.axes.axis('off')

        self.MplWidget.canvas.draw()

    def setupUi(self, MenuOutput):
        MenuOutput.setObjectName("MenuOutput")
        MenuOutput.setFixedSize(673, 435)
        MenuOutput.setWindowIcon(QtGui.QIcon('icon.png'))
        self.centralwidget = QtWidgets.QWidget(MenuOutput)
        self.centralwidget.setObjectName("centralwidget")
        self.labelTitle2 = QtWidgets.QLabel(self.centralwidget)
        self.labelTitle2.setGeometry(QtCore.QRect(170, 10, 363, 38))
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(21)
        font.setBold(True)
        font.setWeight(75)
        font.setStrikeOut(False)
        font.setKerning(True)
        self.labelTitle2.setFont(font)
        self.labelTitle2.setAlignment(QtCore.Qt.AlignCenter)
        self.labelTitle2.setObjectName("labelTitle2")
        self.MplWidget = MplWidget(self.centralwidget)
        self.MplWidget.setGeometry(QtCore.QRect(20, 70, 331, 321))
        self.MplWidget.setObjectName("MplWidget")
        self.frameNilaiSoma = QtWidgets.QFrame(self.centralwidget)
        self.frameNilaiSoma.setGeometry(QtCore.QRect(380, 90, 271, 211))
        self.frameNilaiSoma.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.frameNilaiSoma.setFrameShadow(QtWidgets.QFrame.Raised)
        self.frameNilaiSoma.setObjectName("frameNilaiSoma")
        self.gridLayout = QtWidgets.QGridLayout(self.frameNilaiSoma)
        self.gridLayout.setObjectName("gridLayout")
        self.labelMesoResult = QtWidgets.QLabel(self.frameNilaiSoma)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelMesoResult.setFont(font)
        self.labelMesoResult.setObjectName("labelMesoResult")
        self.gridLayout.addWidget(self.labelMesoResult, 3, 1, 1, 1)
        self.labelMesomorph = QtWidgets.QLabel(self.frameNilaiSoma)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(12)
        self.labelMesomorph.setFont(font)
        self.labelMesomorph.setObjectName("labelMesomorph")
        self.gridLayout.addWidget(self.labelMesomorph, 3, 0, 1, 1)
        self.labelEctoResult = QtWidgets.QLabel(self.frameNilaiSoma)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelEctoResult.setFont(font)
        self.labelEctoResult.setObjectName("labelEctoResult")
        self.gridLayout.addWidget(self.labelEctoResult, 4, 1, 1, 1)
        self.labelType = QtWidgets.QLabel(self.frameNilaiSoma)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(12)
        self.labelType.setFont(font)
        self.labelType.setObjectName("labelType")
        self.gridLayout.addWidget(self.labelType, 5, 0, 1, 1)
        self.labelEndoResult = QtWidgets.QLabel(self.frameNilaiSoma)
        self.labelEndoResult.setMinimumSize(QtCore.QSize(148, 0))
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelEndoResult.setFont(font)
        self.labelEndoResult.setObjectName("labelEndoResult")
        self.gridLayout.addWidget(self.labelEndoResult, 2, 1, 1, 1)
        self.labelSubtitle2 = QtWidgets.QLabel(self.frameNilaiSoma)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.labelSubtitle2.setFont(font)
        self.labelSubtitle2.setAlignment(QtCore.Qt.AlignCenter)
        self.labelSubtitle2.setObjectName("labelSubtitle2")
        self.gridLayout.addWidget(self.labelSubtitle2, 1, 0, 1, 2)
        self.labelEctomorph = QtWidgets.QLabel(self.frameNilaiSoma)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(12)
        self.labelEctomorph.setFont(font)
        self.labelEctomorph.setObjectName("labelEctomorph")
        self.gridLayout.addWidget(self.labelEctomorph, 4, 0, 1, 1)
        self.labelEndomorph = QtWidgets.QLabel(self.frameNilaiSoma)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(12)
        self.labelEndomorph.setFont(font)
        self.labelEndomorph.setObjectName("labelEndomorph")
        self.gridLayout.addWidget(self.labelEndomorph, 2, 0, 1, 1)
        self.labelTypeResult = QtWidgets.QLabel(self.frameNilaiSoma)
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(14)
        self.labelTypeResult.setFont(font)
        self.labelTypeResult.setObjectName("labelTypeResult")
        self.gridLayout.addWidget(self.labelTypeResult, 5, 1, 1, 1)
        self.buttonBack = QtWidgets.QPushButton(self.centralwidget)
        self.buttonBack.setGeometry(QtCore.QRect(380, 320, 271, 41))
        self.buttonBack.setMinimumSize(QtCore.QSize(0, 40))
        font = QtGui.QFont()
        font.setFamily("Open Sans")
        font.setPointSize(10)
        font.setBold(False)
        font.setWeight(50)
        self.buttonBack.setFont(font)
        self.buttonBack.setObjectName("buttonBack")
        self.buttonBack.clicked.connect(self.openMenuInput)
        MenuOutput.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(MenuOutput)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 673, 21))
        self.menubar.setObjectName("menubar")
        MenuOutput.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(MenuOutput)
        self.statusbar.setObjectName("statusbar")
        MenuOutput.setStatusBar(self.statusbar)

        self.retranslateUi(MenuOutput)
        QtCore.QMetaObject.connectSlotsByName(MenuOutput)
        self.drawSomatochart()

    def retranslateUi(self, MenuOutput):
        _translate = QtCore.QCoreApplication.translate
        MenuOutput.setWindowTitle(_translate("MenuOutput", "Output Menu"))
        self.labelTitle2.setText(_translate("MenuOutput", "Somatotype Result"))
        self.labelMesoResult.setText(_translate("MenuOutput", ": "+str(round(mesomorph,2))))
        self.labelMesomorph.setText(_translate("MenuOutput", "Mesomorph"))
        self.labelEctoResult.setText(_translate("MenuOutput", ": "+str(round(ectomorph,2))))
        self.labelType.setText(_translate("MenuOutput", "Type"))
        self.labelEndoResult.setText(_translate("MenuOutput", ": "+str(round(endomorph,2))))
        self.labelSubtitle2.setText(_translate("MenuOutput", "Somatotype Score"))
        self.labelEctomorph.setText(_translate("MenuOutput", "Ectomorph"))
        self.labelEndomorph.setText(_translate("MenuOutput", "Endomorph"))
        self.labelTypeResult.setText(_translate("MenuOutput", ": "+dispJenisSoma))
        self.buttonBack.setText(_translate("MenuOutput", "Back"))



if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    menuInput = QtWidgets.QMainWindow()
    ui = Ui_menuInput()
    ui.setupUi(menuInput)
    menuInput.show()
    sys.exit(app.exec_())
